pilot/pkg/model/context.go

```diff
@@ -127,18 +115,39 @@ func (node *Proxy) GetProxyVersion() (string, bool) {
 	return version, found
 }
 
-// GetGatewayMode returns the mode in which the gateway is operating.
-func (node *Proxy) GetGatewayMode() GatewayMode {
-	if modestr, found := node.Metadata["ISTIO_GATEWAY_MODE"]; found {
-		mode := GatewayMode(modestr)
-		switch mode {
-		case MulticlusterGateway:
-			return MulticlusterGateway
-		default:
-			return StandardGateway
+// GetProxyMTLSInclusionSet returns list of services for which Istio mTLS should be enabled
+// if there is a global destination rule specifying ISTIO_MUTUAL as the tls mode for all
+// services in the system. The value is obtained from ISTIO_META_ISTIO_MTLS_INCLUDE_ONLY
+// metadata variable provided by the sidecar proxy.
+//
+// We need this because DestinationRules are typically specified by the service owner but
+// applied at all clients. One or more clients may not be able to to mTLS. So these clients
+// need an escape hatch. On the server side, a permissive mTLS mode will allow both TLS and
+// non-TLS traffic.
+func (node *Proxy) GetProxyMTLSInclusionSet() map[string]bool {
+	mTLSInclusionMap := make(map[string]bool)
+	if list, found := node.Metadata["ISTIO_MTLS_INCLUDE_ONLY"]; found {
+		for _, s := range strings.Split(list, ",") {
+			mTLSInclusionMap[s] = true
 		}
+		// NOTE: if the metadata is set, but has no entries i.e. "",
+		// it implies do not do mTLS for anything
+		return mTLSInclusionMap
+	}
+
+	// If the metadata is not set, then do whatever is dictated by the destination rule.
+	return nil
+}
+
+// HasSnowFlakes returns true if this proxy has any configuration that could cause it to
+// receive xDS output that is different from other proxies.
+// Currently, we check for things like mtls inclusion list which is used only by the
+// multi-cluster gateway.
+func (node *Proxy) HasSnowFlakes() bool {
+	if _, found := node.Metadata["ISTIO_MTLS_INCLUDE_ONLY"]; found {
```

Comment:
should be ISTIO_MTLS_INCLUSION_ONLY to be compliant with it in values.yaml

Reply:
Oh good catch